### PR TITLE
Switch default build JDK to Java 12.

### LINKED ISSUE #2141: Raise build JDK requirement to Java 12
We now have travis infrastructure for installing multiple JDKs, including ones not officially suported by travis. This opens up the possibility of raising our minimum JDK version to build Armeria to Java 12. This makes it simpler to work on supporting newer APIs like in #2119 

Shall we raise the requirement?

### PR DESCRIPTION
**Pull request by anuraaga**
Fixes #2141

### CODE REVIEW

**Comment by trustin** [comment]
Could you also update the `release` task in `/build.gradle` to require Java 12? Maybe should we make the entire build require Java 12 rather than just the release process?

**Comment by anuraaga** [comment]
Thanks indeed. Changed the release task to require Java 12, and added a check for normal compile that verifies >=12 (don't think we have to require exactly 12 for normal building during development)

File: build.gradle [hunk 1]
```diff
@@ -45,6 +45,16 @@ allprojects {
     }
 }
 
+task checkJavaVersion {
+    doLast {
+        if (JavaVersion.current().majorVersion.toInteger() < 12) {
+            throw new GradleException("The java version used ${JavaVersion.current()} is too old to " +
```
Comment (trustin) [inline build.gradle]: - java -> Java
- How about wrapping `${JavaVersion.current()}` with parens?

File: build.gradle [hunk 2]
```diff
@@ -146,7 +156,7 @@ allprojects {
 
 // Require to use JDK 11 when releasing.
```
Comment (minwoox) [inline build.gradle]: nit: JDK 12 😄

**Pull request merged**